### PR TITLE
[HLD] Packet I/O TX path - Send to Ingress HLD

### PR DESCRIPTION
**Pull request by baxia-lan**
Create new PR since https://github.com/sonic-net/SONiC/pull/986 was closed by accident

### CODE REVIEW

**Comment by baxia-lan** [comment]
@prsunny Hi Prince, could you please take a look at this HLD?

File: doc/pins/send_to_ingress_hld.md [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+      "alias": "NoAliasEth0/1",
+      "index": "1",
+      "lanes": "9,10,11,12,13,14,15,16",
+      "mtu": "9100",
+      "speed": "400000"
+    },
+    …
+  },
+  "SEND_TO_INGRESS_PORT": {
+     "SEND_TO_INGRESS": {}
```
Comment (prsunny) [inline doc/pins/send_to_ingress_hld.md:90]: Can you please provide the full info here? What is expected in the `{}`

File: doc/pins/send_to_ingress_hld.md [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+strncpy(attr.value.chardata, send_to_ingress_name.c_str(), size - 1);
+attr.value.chardata[size - 1] = '\0';
+ingress_attribs.push_back(attr);
+// If this isn't passed in true, the false setting makes
+// the device unready for later attempts to set UP/RUNNING
+attr.id = SAI_HOSTIF_ATTR_OPER_STATUS;
+attr.value.booldata = true;
+ingress_attribs.push_back(attr);
+// Get CPU port object id to signal send to ingress
+attr.id = SAI_SWITCH_ATTR_CPU_PORT;
```
Comment (prsunny) [inline doc/pins/send_to_ingress_hld.md:117]: Why would CPU port be provided here? Is it used as an indication for SEND to ingress?

File: doc/pins/send_to_ingress_hld.md (hunk 1 above)
Comment (baxia-lan) [reply doc/pins/send_to_ingress_hld.md:90]: No attributes are needed currently. Only port name, `SEND_TO_INGRESS`, is needed as the configurable port name, and netdev name.

File: doc/pins/send_to_ingress_hld.md (hunk 2 above)
Comment (baxia-lan) [reply doc/pins/send_to_ingress_hld.md:117]: Yes. P4Runtime will create a socket for the “send_to_ingress” netdev port and write into this socket for packets marked to send to “send_to_ingress”. Packets egressing out of the socket end up as incoming packets on the CPU port of the switch.

Ref https://github.com/sonic-net/SONiC/blob/master/doc/pins/Packet_io.md#transmit-path-design

**Review by prsunny — APPROVED** [review]

**Pull request merged**